src/daemon.rs

```diff
@@ -542,7 +503,66 @@ pub fn daemon_init(settings: Settings, config: Config) -> Result<Daemon, Error>
         daemon.cpus.push(cpu);
     }
 
-    Ok(daemon)
+    let daemon_mutex = Arc::new(Mutex::new(daemon));
+    let c_daemon_mutex = Arc::clone(&daemon_mutex);
+
+    listen("/tmp/acs.sock", c_daemon_mutex);
+
+    Ok(daemon_mutex)
+}
+
+pub fn run(daemon_mutex: Arc<Mutex<Daemon>>) -> Result<(), Error> {
+    // Aquire the lock for a bit
+    let mut daemon = daemon_mutex.lock().unwrap();
+
+    daemon.init();
+
+    if daemon.settings.testing {
+        // Choose which mode acs runs in
+        if daemon.settings.edit {
+            let mut reps = 4;
+            while reps > 0 {
+                daemon.single_edit()?;
+                reps -= 1;
+            }
+        } else {
+            let mut reps = 4;
+            while reps > 0 {
+                daemon.single_monit()?;
+                reps -= 1;
+            }
+        }
+    } else {
+        // Before runnig the loop drop the lock and aquire it again later within the loop
+        let mode = daemon.settings.edit.clone();
+
+        let effective_timeout = if daemon.charging {
+            daemon.timeout.clone()
+        } else {
+            daemon.timeout_battery.clone()
+        };
+
+        drop(daemon);
+
+        // Choose which mode acs runs in
+        if mode {
+            loop {
+                let mut daemon = daemon_mutex.lock().unwrap();
```

Comment:
I am a little worried about the speed of this

Reply:
I guess we have a sleep right after it, maybe it does not matter here

Reply:
It doesn't matter much, the daemon is only "locked" for an incredibly small amount of time while data is being read.